### PR TITLE
Fixed #24493 - Added setdefault method to BaseContext.

### PR DESCRIPTION
**Pull request by prestontimmons**

### CODE REVIEW

File: docs/ref/templates/api.txt [hunk 1]
```diff
@@ -523,6 +523,18 @@ against ``dict``::
                 'update': 'value',
             })
 
+.. method:: Context.get(key, otherwise=None)
+
+    Returns the value for ``key`` if ``key`` is in the context, else returns
+    ``otherwise``.
+
+.. versionadded:: 1.9
```
Comment (timgraham) [inline docs/ref/templates/api.txt]: this goes under`.. method:: Context.setdefault(key, default=None)` (indented)

File: docs/ref/templates/api.txt [hunk 2]
```diff
@@ -523,6 +523,18 @@ against ``dict``::
                 'update': 'value',
             })
 
+.. method:: Context.get(key, otherwise=None)
```
Comment (timgraham) [inline docs/ref/templates/api.txt]: Ideally this doc would be a separate commit so it could be backported.

File: docs/releases/1.9.txt [hunk 3]
```diff
@@ -166,6 +166,9 @@ Templates
   helper can now store results in a template variable by using the ``as``
   argument.
 
+* Added a :meth:`~django.template.Context.setdefault` method to
```
Comment (timgraham) [inline docs/releases/1.9.txt]: or more simply:
```
Added a `Context.setdefault() <django.template.Context.setdefault>` method.
```

File: tests/template_tests/test_context.py [hunk 4]
```diff
@@ -30,6 +30,17 @@ def test_context(self):
             self.assertEqual(c['a'], 3)
         self.assertEqual(c['a'], 1)
 
+    def test_setdefault(self):
+        c = Context()
+
+        x = c.setdefault("x", 42)
```
Comment (timgraham) [inline tests/template_tests/test_context.py]: single quotes please

**Comment by prestontimmons** [comment]
Thanks, Tim. This is updated. I also moved the position of the new method docs. I think it fits better moved up where the dict-like behavior of context is explained.

There was a docs build error I can't reproduce. The jenkins console output looks like it wasn't related to the code change. Is there a way to rerun it?

**Comment by timgraham** [comment]
Not sure why the docs build sometimes does that. merged in 4d9414098bd98aacd241827a01f6ad2dff71f113 and 388e79e9fcbf689d5708134534e55d176cfbcee3, thanks!
